### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

**Review by olbjo — APPROVED** [review]

File: doc/compliancetest.rst [hunk 1]
```diff
@@ -203,6 +203,31 @@ left of the "PowerOutlet" text in the tool bar. The symbol to the right of the
 "PowerOutlet" text shows a green check mark when the outputs are on, and a
 black cross when the outputs are off (or when the power outlet not is connected).
 
+Hardware naming
+^^^^^^^^^^^^^^^
+For the conformance test is different type of Siemens hardware used.
```
Comment (hefloryd) [inline doc/compliancetest.rst:208]: This sentence structure is weird. "Different types of Siemens hardware are used for the conformance test."

File: doc/compliancetest.rst [hunk 2]
```diff
@@ -324,7 +346,7 @@ For multi-port devices:
 For conformance class B:
 
 * Topology discovery check
-* Non-Profinet neighbour setup
+* Topology non-PN  (Use non-Profinet-neighbor setup)
```
Comment (hefloryd) [inline doc/compliancetest.rst:349]: We use British English spelling. Neighbour.

File: doc/getting_started_rtkernel.rst [hunk 3]
```diff
@@ -243,8 +245,8 @@ The flash usage is text+data, as the RAM initialization values are stored in fla
 
 Run tests on XMC4800 target
 ---------------------------
-In order to compile the test code, enable BUILD_TESTING and disable TEST_DEBUG
-in cmake. Reduce PNET_MAX_FILENAME_LENGTH to 30 bytes.
+In order to compile the test code, enable BUILD_TESTING and disable TEST_DEBUG in
```
Comment (hefloryd) [inline doc/getting_started_rtkernel.rst:248]: Isn't this the default (BUILD_TESTING=ON, TEST_DEBUG=OFF)?

File: doc/network_topology_detection.rst [hunk 4]
```diff
@@ -21,6 +21,10 @@ SNMP
 ----
 It is possible to ask an IO-Controller or IO-device (conformance class B or
 higher) about its neighbours, using the Simple Network Management Protocol (SNMP).
+
+SNMP agents are collecting information on individual devices, and the SNMP manager
```
Comment (hefloryd) [inline doc/network_topology_detection.rst:25]: SNMP agents collect information, SNMP manager retrieves information

File: doc/profinet_basics.rst [hunk 5]
```diff
@@ -96,6 +96,37 @@ Real Time Class:
...
+   |    Ethernet Switch    |
+   |                       |
+   +----------+------------+
+              |
+              |
+              |
+     Profinet interface X1
+
+The device has one interface and two ports.
+For Profinet conformance class B and higher, there are three MAC addresses assigned:
```
Comment (hefloryd) [inline doc/profinet_basics.rst:119]: there are three assigned MAC addresses

File: doc/tutorial.rst [hunk 6]
```diff
@@ -146,6 +146,18 @@ See the "Real-time properties of Linux" page in this document for solutions, and
 the "Using Codesys soft PLC" page for workarounds.
 
 
+Trouble shooting
```
Comment (hefloryd) [inline doc/tutorial.rst:149]: Troubleshooting (or Trouble-shooting)

File: doc/getting_started_rtkernel.rst (hunk 3 above)
Comment (pyhys) [reply doc/getting_started_rtkernel.rst:248]: Changed to "In order to compile the test code, make sure to use BUILD_TESTING and that TEST_DEBUG is disabled. Set PNET_MAX_AR to 1, and reduce PNET_MAX_FILENAME_LENGTH to 30 bytes."

File: doc/network_topology_detection.rst (hunk 4 above)
Comment (pyhys) [reply doc/network_topology_detection.rst:25]: Changed to "SNMP agents collect information on individual devices, and the SNMP manager retrieves information from them."

File: doc/profinet_basics.rst (hunk 5 above)
Comment (pyhys) [reply doc/profinet_basics.rst:119]: Done

File: doc/tutorial.rst (hunk 6 above)
Comment (pyhys) [reply doc/tutorial.rst:149]: Done

File: doc/compliancetest.rst (hunk 2 above)
Comment (pyhys) [reply doc/compliancetest.rst:349]: Done

File: doc/compliancetest.rst (hunk 1 above)
Comment (pyhys) [reply doc/compliancetest.rst:208]: Done

**Pull request merged**